packages/teleterm/src/services/grpcCerts/makeCert/makeCert.ts

```diff
@@ -0,0 +1,198 @@
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ Copyright 2022 Gravitational, Inc.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+import { pki, md, util, random } from 'node-forge';
+import { promisify } from 'util';
+
+const generateKeyPair = promisify(pki.rsa.generateKeyPair.bind(pki.rsa));
+
+interface GeneratedCert {
+  key: string;
+  cert: string;
+}
+
+export async function createCA({
+  organization,
+  countryCode,
+  state,
+  locality,
+  validityDays,
+}: {
+  organization: string;
+  countryCode: string;
+  state: string;
+  locality: string;
+  validityDays: number;
+}): Promise<GeneratedCert> {
+  // certificate Attributes: https://git.io/fptna
+  const attributes = [
+    { name: 'commonName', value: organization },
+    { name: 'countryName', value: countryCode },
+    { name: 'stateOrProvinceName', value: state },
+    { name: 'localityName', value: locality },
+    { name: 'organizationName', value: organization },
+  ];
+
+  // required certificate extensions for a certificate authority
+  const extensions = [
+    { name: 'basicConstraints', cA: true, critical: true },
+    {
+      name: 'keyUsage',
+      keyCertSign: true,
+      critical: true,
+    },
+  ];
+
+  return await generateCert({
+    subject: attributes,
+    issuer: attributes,
+    extensions,
+    validityDays,
+  });
+}
+
+export async function createCert({
+  domains,
+  validityDays,
+  caKey,
+  caCert,
+}: {
+  domains: string[];
+  validityDays: number;
+  caKey: string;
+  caCert: string;
+}): Promise<GeneratedCert> {
+  // certificate Attributes: https://git.io/fptna
+  const attributes = [
+    { name: 'commonName', value: domains[0] }, // use the first address as common name
+  ];
+
+  // required certificate extensions for a tls certificate
+  // keyUsage has been extended by entries from https://github.com/digitalbazaar/forge#x509
+  const extensions = [
+    { name: 'basicConstraints', cA: false, critical: true },
+    {
+      name: 'keyUsage',
+      digitalSignature: true,
+      keyEncipherment: true,
+      critical: true,
+    },
+    { name: 'extKeyUsage', serverAuth: true, clientAuth: true },
+    {
+      name: 'subjectAltName',
+      altNames: domains.map(domain => {
+        const types = { domain: 2, ip: 7 }; // available Types: https://git.io/fptng
+        const ipRegex = /^(?!0)(?!.*\.$)((1?\d?\d|25[0-5]|2[0-4]\d)(\.|$)){4}$/;
```

Comment:
Nit: I'd recommend moving it to `const IP_REGEX` somewhere at the top of the file.

Reply:
Done